Source/FORMData.m

```diff
@@ -806,71 +791,71 @@ - (NSArray *)updateTargets:(NSArray *)targets {
     NSMutableArray *updatedIndexPaths = [NSMutableArray new];
 
     for (FORMTarget *target in targets) {
-
         BOOL shouldContinue = (![self evaluateCondition:target.condition] ||
                                target.type == FORMTargetTypeSection ||
                                (self.hiddenFieldsAndFieldIDsDictionary)[target.targetID]);
-
         if (shouldContinue) {
             continue;
         }
 
-
         __block FORMField *field = nil;
+        [self fieldWithID:target.targetID
+    includingHiddenFields:YES
+               completion:^(FORMField *foundField, NSIndexPath *indexPath) {
+                   if (foundField) {
+                       field = foundField;
+                       if (indexPath) {
+                           [updatedIndexPaths addObject:indexPath];
+                       }
+                   }
+               }];
 
-        [self fieldWithID:target.targetID includingHiddenFields:YES completion:^(FORMField *foundField, NSIndexPath *indexPath) {
-            if (foundField) {
-                field = foundField;
-                if (indexPath) {
-                    [updatedIndexPaths addObject:indexPath];
+        if (field) {
+            NSArray *properties = [target propertiesToUpdate];
+            for (NSString *propertyName in properties) {
+                id value = [target valueForKey:propertyName];
+
+                if (![propertyName isEqualToString:@"value"]) {
+                    [field setValue:value forKey:propertyName];
```

Comment:
This is where the magic of this PR happens, here the target properties get set to the field